### PR TITLE
Some UI & RESTful Endpoint Refactoring

### PR DESCRIPTION
**Pull request by maknoon**
**related issues**: #30  
**changes**:  
- in `airplane.py`:  
  - refactored `/customer`, `/ff`, `/airport`, `/gate`, `/schedule`, `/flight`, `/itinerary`  
  - added some behavior for an admin view to be rendered via Jinja templating  
  - added helper function for wrapping some values from `PATCH` requests  
- in `airplanedb.py`:  
  - fixed a bunch of things  
  - renamed all returned JSON props to proper snake_case format
- in `templates/db.html`:  
  - implemented an example for how the admin UI can be rendered from a JSON object passed into `render_template()`  
  screen shot 2017-11-15 at 11 57 42 pm  

  - implemented a similar example for the user UI  
  screen shot 2017-11-16 at 3 11 39 am  

- other:
  - removed redundant chunk of documentation from `README`  

**how to test**: use a REST client as [documented](https://github.com/maknoon/airdb/wiki/Using-a-REST-Client-to-test-our-API) in the wiki. try logging in as a `user` or an `admin` (use `/logout` to logout)

### CODE REVIEW

**Review by jefftwin — APPROVED** [review]

File: airplanedb.py [hunk 1]
```diff
@@ -869,28 +911,20 @@ def get_airport(self, ap_id):
...
-                    }
-                    dataList.append(ap_object)
             else:
                 airports = cursor.fetchone()
+            for airport in airports:
                 ap_object = {
-                    'ID': airports[0],
-                    'City': airports[1],
-                    'Country': airports[2]
+                    'airport_id': airport[0],
```
Comment (jefftwin) [inline airplanedb.py]: @maknoon  does this implementation give the desired return? When I tested this on my branch, cursor.fetchone() returns a non-iteratable object so saying for airport in airports for cursor.fetchone() returns a corrupt json object

`else:
    airports = cursor.fetchone()

for airport in airports:`

**Review by jefftwin — CHANGES_REQUESTED** [review]

File: airplane.py [hunk 2]
```diff
@@ -60,160 +65,239 @@ def reset():
...
 
-    return data
+    # fetch a airport by id
+    if request.method == 'GET':
+        airport = airdb.get_airport(apid)
+        if airport == 0: abort(404)
+        else:
+            airport_json = {'airport_id':apid,'city':airport[1],
+                            'country':airport[2]}
+            res_body = json.dumps(airport_json, indent=4, separators=(',', ': '))
```
Comment (jefftwin) [inline airplane.py]: hmmmm i think we should just json.dump the return object from airdb.get_airport. this is return corrupt json object

**Review by jefftwin — CHANGES_REQUESTED** [review]

File: airplanedb.py (hunk 1 above)
Comment (maknoon) [reply airplanedb.py]: You're right; `json.dumps` won't work with how I am constructing the JSON object for `res_body` in the `/airport` router function. It's not a corrupt JSON object; rather that my function is taking values at indices of a string that is generated by `json.dumps`. ref. [json.dumps in Python](https://docs.python.org/2/library/json.html#json.dumps)

**Review by jefftwin — APPROVED** [review]

File: airplane.py (hunk 2 above)
Comment (maknoon) [reply airplane.py]: even better would be to just return what the db gives us! 😉

**Review by katieli16 — APPROVED** [review]
Also fixed the commented-out FF update

**Review by galenhwang — APPROVED** [review]

**Pull request merged**